cmd/tests/tracing_module_test.go

```diff
@@ -128,6 +98,229 @@ func TestTracingClient_DoesNotInterfereWithHTTPModule(t *testing.T) {
+
+	jsonResults, err := afero.ReadFile(ts.FS, "results.json")
+	require.NoError(t, err)
+
+	assertHasTraceIDMetadata(t, jsonResults)
+}
+
+func TestTracingInstrumentHTTP_FillsParams(t *testing.T) {
+	t.Parallel()
+	tb := httpmultibin.NewHTTPMultiBin(t)
+
+	var gotRequests int64
+
+	tb.Mux.HandleFunc("/tracing", func(w http.ResponseWriter, r *http.Request) {
+		atomic.AddInt64(&gotRequests, 1)
+
+		assert.NotEmpty(t, r.Header.Get("traceparent"))
+		assert.Len(t, r.Header.Get("traceparent"), 55)
+
+		assert.NotEmpty(t, r.Header.Get("X-Test-Header"))
+		assert.Equal(t, "test", r.Header.Get("X-Test-Header"))
+	})
+
+	script := tb.Replacer.Replace(`
+		import http from "k6/http";
+		import { check } from "k6";
+		import tracing from "k6/experimental/tracing";
+
+		tracing.instrumentHTTP({
+			propagator: "w3c",
+		})
+
+		const testHeaders = {
+			"X-Test-Header": "test",
+		}
+
+		export default function () {
+			http.del("HTTPBIN_IP_URL/tracing", null, { headers: testHeaders });
+			http.get("HTTPBIN_IP_URL/tracing", { headers: testHeaders });
+			http.head("HTTPBIN_IP_URL/tracing", { headers: testHeaders });
+			http.options("HTTPBIN_IP_URL/tracing", null, { headers: testHeaders });
+			http.patch("HTTPBIN_IP_URL/tracing", null, { headers: testHeaders });
+			http.post("HTTPBIN_IP_URL/tracing", null, { headers: testHeaders });
+			http.put("HTTPBIN_IP_URL/tracing", null, { headers: testHeaders });
+			http.request("GET", "HTTPBIN_IP_URL/tracing", null, { headers: testHeaders });
+		};
+	`)
+
+	ts := getSingleFileTestState(t, script, []string{"--out", "json=results.json"}, 0)
+	cmd.ExecuteWithGlobalState(ts.GlobalState)
+
+	assert.Equal(t, int64(8), atomic.LoadInt64(&gotRequests))
+
+	jsonResults, err := afero.ReadFile(ts.FS, "results.json")
+	require.NoError(t, err)
+
+	assertHasTraceIDMetadata(t, jsonResults)
+}
+
+func TestTracingInstrumentHTTP_FailsOutsideOfInitContext(t *testing.T) {
+	t.Parallel()
+	tb := httpmultibin.NewHTTPMultiBin(t)
+
+	script := tb.Replacer.Replace(`
+		import tracing from "k6/experimental/tracing";
+
+		export default function() {
+			tracing.instrumentHTTP({
+				propagator: "w3c",
+			})
+		}
+	`)
+
+	ts := getSingleFileTestState(t, script, []string{}, 0)
+	ts.ExpectedExitCode = 0
+	cmd.ExecuteWithGlobalState(ts.GlobalState)
+
+	assert.Contains(t, ts.Stderr.String(), "instrumentHTTP can only be called in the init context")
+}
+
+func TestTracingInstrumentHTTP_CannotBeCalledTwice(t *testing.T) {
+	t.Parallel()
+	tb := httpmultibin.NewHTTPMultiBin(t)
+
+	script := tb.Replacer.Replace(`
+		import tracing from "k6/experimental/tracing";
+
+		tracing.instrumentHTTP({
+			propagator: "w3c",
+		})
+
+		tracing.instrumentHTTP({
+			propagator: "w3c",
+		})
+
+		export default function() {
+		}
+	`)
+
+	ts := getSingleFileTestState(t, script, []string{}, 0)
+	ts.ExpectedExitCode = 107
+	cmd.ExecuteWithGlobalState(ts.GlobalState)
+
+	assert.Contains(t, ts.Stderr.String(), "instrumentHTTP can only be called once")
+}
```

Comment:
I kind of prefer if those are not integration tests as they are more expensive and in this case there is no need for them IMO.

The functionality tested is pretty common :
1. exception in an iteration does not abort the whole test
2. an exception in the init context - does.

Reply:
Let me see if I can easily turn them into standard unit tests indeed. I've struggled with `instrumentHTTP` depending on require in the past, but with the changes with made, it might not be such an issue anymore 👍🏻